app/cdap/components/hydrator/components/TopPanel/NameAndDescription.tsx

```diff
@@ -0,0 +1,169 @@
+import React, { useState } from 'react';
+import styled from 'styled-components';
+
+interface INameAndDescriptionProps {
+  metadataExpanded?: boolean;
+  state?: any;
+  invalidName?: string;
+  parsedDescription?: string;
+  saveMetadata?: (event: any) => void;
+  resetMetadata?: (event: any) => void;
+}
+
+const PipelineName = styled.div`
+  color: gray;
+  font-size: 14px;
+  margin-top: 5px;
+  line-height: 17px;
+
+  ${({ errorName }) =>
+    errorName &&
+    `
+    color: #d15668;
+  `}
+`;
+
+const PipelineDescription = styled.div`
+  color: gray;
+  font-size: 12px;
+`;
+
+const useStyle = makeStyle(() => {
+  return {
+    arrow: {
+      color: '#d15668',
+    },
+    tooltip: {
+      fontSize: '13px',
+    },
+    dangerTooltip: {
+      fontSize: '13px',
+      background: '#d15668',
+    },
+  };
+});
+
+export const NameAndDescription = ({
+  metadataExpanded,
+  state,
+  invalidName,
+  parsedDescription,
+  saveMetadata,
+  resetMetadata,
+}: INameAndDescriptionProps) => {
+  const [name, setName] = useState(state.metadata.name);
+  const [description, setDescription] = useState(state.metadata.description);
+  const handleNameChange = (e) => {
+    setName(e.target.value);
+    state.metadata.name = e.target.value;
+  };
+  const handleDescriptionChange = (e) => {
+    setDescription(e.target.value);
+    state.metadata.description = e.target.value;
+  };
+  const onEnterOnMetadata = (event) => {
+    // Save when user hits ENTER key.
+    if (event.nativeEvent.keyCode === 13) {
+      saveMetadata(event);
+    } else if (event.nativeEvent.keyCode === 27) {
+      // Reset if the user hits ESC key.
+      resetMetadata(event);
+    }
+  };
+  const classes = useStyle();
+
+  return (
+    <div>
+      {!metadataExpanded && (
+        <div>
+          {invalidName ? (
+            <Tooltip
+              title={'Invalid Name'}
+              arrow
+              open
+              placement="bottom-start"
+              classes={{ arrow: classes.arrow, tooltip: classes.dangerTooltip }}
+            >
+              <PipelineName errorName={true}>{name || 'Name your pipeline'}</PipelineName>
+            </Tooltip>
+          ) : (
+            <Tooltip
+              title={name}
+              arrow
+              placement="bottom-start"
+              classes={{ tooltip: classes.tooltip }}
+            >
+              <PipelineName errorName={false}>{name || 'Name your pipeline'}</PipelineName>
+            </Tooltip>
+          )}
+          <Tooltip
+            title={parsedDescription}
+            arrow
+            placement="bottom-start"
+            classes={{ tooltip: classes.tooltip }}
+          >
+            <PipelineDescription>
+              {parsedDescription || 'Enter a description for your pipeline'}
+            </PipelineDescription>
+          </Tooltip>
+        </div>
+      )}
+      {metadataExpanded && (
+        <div>
+          <input
```

Comment:
use material ui versions of all these html elements

Reply:
After some trying, found that the input and textarea styles are tangled deeply in the toppanel bootstrap styling, decided to leave it for now, and will address them all together after the whole top panel is migrated.